src/canvas/helper.ts

```diff
@@ -46,12 +63,18 @@ export function createRadialGradient(
     let x = obj.x == null ? 0.5 : obj.x;
     let y = obj.y == null ? 0.5 : obj.y;
     let r = obj.r == null ? 0.5 : obj.r;
+
     if (!obj.global) {
         x = x * width + rect.x;
         y = y * height + rect.y;
         r = r * min;
     }
 
+    if (!isSafeNum(x) || !isSafeNum(y) || !isSafeNum(r, true)) {
+        console.warn('The provided value is non-finite. You must provide x and y.');
+        return;
```

Comment:
return nothing here is not safe. Because we expect this method to return a CanvasGradient and set it to CanvasRenderingContext. If we return `null`, it will be an undefined behavior.

I think we can fallback to default value when the `x`, `y`, `r` is invalid like `createLinearGradient` did.

Also there are several concerns about this warn message being to frequently:
1. It should be warn only once if the `CanvasGradientObject` is not changed. Use `WeakMap` to mark this `CanvasGradientObject` has been warned may be a solution.
2. The `boundingRect` sometimes may be infinite or NaN(when we don't want to show the element) and leading this `x`, `y`, `r` to be invalid. In this case, the error message will not be helpful but confuse the 
developers. So it's better to check the `obj.x`, `obj.y`, `obj.r` earlier and give warning message. Then we do check `x`, `y`, `r` again and fallback to default value.
3. The check of `obj.x`, `obj.y`, `obj.r` and warning message can be done only when `process.env.NODE_ENV !== 'production'` so these can be ommitted in the production environment.

At last the `createLinearGradient` should do the same.

Reply:
The obj is changed when mouse hover, can not use WeakMap to add warning log only once. So I added a guard to ensure that the parameters are valid.

![image](https://user-images.githubusercontent.com/8437666/159258466-839e277e-56e8-4d2e-afac-a57a8b7f930b.png)